File: alura/entity_framework_core/Store/Store.Cli/Models/Product.cs
namespace Store.Cli.Models
{
    public class Product
    {
        public Product()
        {

        }

        public Product(string name, string category)
        {
            Name = name;
            Category = category;
        }

        public int Id { get; set; }

        public string Name { get; set; }

        public string Category { get; set; }

        public override string ToString() => $"Product[Id={Id}, Name={Name}, Category={Category}]";
    }
}

File: alura/course_jpa_003/marketplace/src/main/java/br/com/flaviogf/marketplace/models/Product.java
package br.com.flaviogf.marketplace.models;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.List;

@Entity
@Getter
@Setter
@EqualsAndHashCode(of = "id")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String cover;

    private String description;

    private BigDecimal price;

    @ManyToMany
    private List<Category> categories;

    @ManyToOne
    private Store store;
}
